File: frontend/src/types/project.ts
export interface Project {
  id: number;
  name: string;
  status: string;

  idea?: string | null;
  problem?: string | null;
  audience?: string | null;
  goal?: string | null;
  budget?: string | null;
  timeline?: string | null;

  ai_report?: string | null;
}

File: backend/app/schemas/project.py
from pydantic import BaseModel


class ProjectCreate(BaseModel):
    name: str


class ProjectResponse(BaseModel):
    id: int
    name: str
    status: str

    idea: str | None = None
    problem: str | None = None
    audience: str | None = None
    goal: str | None = None
    budget: str | None = None
    timeline: str | None = None

    ai_report: str | None = None

    class Config:
        from_attributes = True